### DISCUSSION TITLE
Code owners

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1233] [General]
Recent discussion on crate smashing and [unsafe](https://doc.rust-lang.org/nomicon/) policy seems both to be promoting the idea that we have among the set of maintainers devs with differing strengths/weaknesses. Is there, or should we be investigating, having different review policy for different parts of the codebase (internally in `rust-bitcoin` and also across the stack of crates)?

Should this be enforced by tooling or just worked out and kept in everyones heads?

For example, I know very little about `unsafe` or PSBTs so my reviews of the former are worthless and the latter are only "code style and adherence to project policy" reviews.

**Comment by Kixunil** [comment]
For `unsafe` I'd definitely want at least two reviewers who understand `unsafe` well. Though maybe if one of them is the contributor it's OK.

**Comment by apoelstra** [comment]
I think it's probably okay for now to keep this sort of information (who understands `unsafe` etc) in peoples' heads. We're not a *lot* of people and nobody tends to claim expertise they don't have.

**Comment by tcharding** [comment]
Seems I open this discussion again in issues: https://github.com/rust-bitcoin/rust-bitcoin/issues/4567
